### PR TITLE
PR06 doc string fixes

### PR DESCRIPTION
**Pull request by tolaa001**
This PR contains doctrine PR06 fixes mostly doing the conversion below(not that many left to fix :)):

boolean to bool

string to str

integer to int

Tests unchanged

black pandas ran successfully

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @tolaa001 . can you merge master. some of these look familiar.

File: pandas/core/computation/eval.py [hunk 1]
```diff
@@ -198,14 +198,14 @@ def eval(
         <https://docs.python.org/3/reference/simple_stmts.html#simple-statements>`__,
         only Python `expressions
         <https://docs.python.org/3/reference/simple_stmts.html#expression-statements>`__.
-    parser : string, default 'pandas', {'pandas', 'python'}
+    parser : str, default 'pandas', {'pandas', 'python'}
```
Comment (simonjayhawkins) [inline pandas/core/computation/eval.py]:
```suggestion
    parser : {'pandas', 'python'}, optional, default 'pandas'
```

File: pandas/core/computation/eval.py [hunk 2]
```diff
@@ -198,14 +198,14 @@ def eval(
...
         The parser to use to construct the syntax tree from the expression. The
         default of ``'pandas'`` parses code slightly different than standard
         Python. Alternatively, you can parse an expression using the
         ``'python'`` parser to retain strict Python semantics.  See the
         :ref:`enhancing performance <enhancingperf.eval>` documentation for
         more details.
-    engine : string or None, default 'numexpr', {'python', 'numexpr'}
+    engine : str or None, default 'numexpr', {'python', 'numexpr'}
```
Comment (simonjayhawkins) [inline pandas/core/computation/eval.py]:
```suggestion
    engine : {'python', 'numexpr'}, optional, default 'numexpr'
```

File: pandas/core/tools/numeric.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def to_numeric(arg, errors="raise", downcast=None):
         - If 'raise', then invalid parsing will raise an exception
         - If 'coerce', then invalid parsing will be set as NaN
         - If 'ignore', then invalid parsing will return the input
-    downcast : {'integer', 'signed', 'unsigned', 'float'} , default None
+    downcast : {'int', 'signed', 'unsigned', 'float'} , default None
```
Comment (simonjayhawkins) [inline pandas/core/tools/numeric.py]: This shouldn't change. probably error in validation. @datapythonista

File: pandas/io/pytables.py [hunk 4]
```diff
@@ -1025,7 +1025,7 @@ def append(
                        Write as a PyTables Table structure which may perform
                        worse but allow more flexible operations like searching
                        / selecting subsets of the data
-        append       : boolean, default True, append the input data to the
+        append       : bool, default True, append the input data to the
```
Comment (simonjayhawkins) [inline pandas/io/pytables.py]: `append the input data to the`  on the next line

File: pandas/io/pytables.py [hunk 5]
```diff
@@ -1037,7 +1037,7 @@ def append(
         chunksize    : size to chunk the writing
         expectedrows : expected TOTAL row size of this table
         encoding     : default None, provide an encoding for strings
-        dropna       : boolean, default False, do not write an ALL nan row to
+        dropna       : bool, default False, do not write an ALL nan row to
```
Comment (simonjayhawkins) [inline pandas/io/pytables.py]: `do not write an ALL nan row to` on the next line

File: pandas/io/stata.py [hunk 6]
```diff
@@ -53,31 +53,31 @@
 )
 
 _statafile_processing_params1 = """\
-convert_dates : boolean, defaults to True
+convert_dates : bool, defaults to True
```
Comment (simonjayhawkins) [inline pandas/io/stata.py]: defaults to True -> default True

File: pandas/io/stata.py [hunk 7]
```diff
@@ -53,31 +53,31 @@
...
     Convert date variables to DataFrame time values.
-convert_categoricals : boolean, defaults to True
+convert_categoricals : bool, defaults to True
```
Comment (simonjayhawkins) [inline pandas/io/stata.py]: same

File: pandas/io/stata.py [hunk 8]
```diff
@@ -53,31 +53,31 @@
...
     Read value labels and convert columns to Categorical/Factor variables."""
 
 _encoding_params = """\
-encoding : string, None or encoding
+encoding : str, None or encoding
     Encoding used to parse the files. None defaults to latin-1."""
 
 _statafile_processing_params2 = """\
-index_col : string, optional, default: None
+index_col : str, optional, default: None
```
Comment (simonjayhawkins) [inline pandas/io/stata.py]: default: None -> default None

File: pandas/io/stata.py [hunk 9]
```diff
@@ -53,31 +53,31 @@
...
     Column to set as index.
-convert_missing : boolean, defaults to False
+convert_missing : bool, defaults to False
```
Comment (simonjayhawkins) [inline pandas/io/stata.py]: optional, default False

File: pandas/io/stata.py [hunk 10]
```diff
@@ -53,31 +53,31 @@
...
     Flag indicating whether to convert missing values to their Stata
     representations.  If False, missing values are replaced with nan.
     If True, columns containing missing values are returned with
     object data types and missing values are represented by
     StataMissingValue objects.
-preserve_dtypes : boolean, defaults to True
+preserve_dtypes : bool, defaults to True
```
Comment (simonjayhawkins) [inline pandas/io/stata.py]: same

File: pandas/io/stata.py [hunk 11]
```diff
@@ -53,31 +53,31 @@
...
     Preserve Stata datatypes. If False, numeric data are upcast to pandas
     default types for foreign data (float64 or int64).
 columns : list or None
     Columns to retain.  Columns will be returned in the given order.  None
     returns all columns.
-order_categoricals : boolean, defaults to True
+order_categoricals : bool, defaults to True
```
Comment (simonjayhawkins) [inline pandas/io/stata.py]: same

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
Thanks @tolaa001 for working on this. A few suggestions.

File: pandas/io/stata.py (hunk 8 above)
Comment (simonjayhawkins) [reply pandas/io/stata.py]: second thoughts no need for default None.

so just `str, optional`

File: pandas/core/computation/eval.py (hunk 1 above)
Comment (datapythonista) [reply pandas/core/computation/eval.py]: The optional is not needed. We just use it when default is `None`, but the `None` means that the parameter is not provided (not the value `None`)

File: pandas/core/tools/numeric.py (hunk 3 above)
Comment (datapythonista) [reply pandas/core/tools/numeric.py]: agree, this seems to refer to the string `integer`, I guess the function does expect `integer` and not `int`. But if you can remove the space before the comma that would be great.

**Review by simonjayhawkins — APPROVED** [review]
@tolaa001 lgtm @datapythonista

File: pandas/core/computation/eval.py [hunk 12]
```diff
@@ -198,14 +198,14 @@ def eval(
...
-    parser : string, default 'pandas', {'pandas', 'python'}
+    parser : {'pandas', 'python'}, default 'pandas'
         The parser to use to construct the syntax tree from the expression. The
         default of ``'pandas'`` parses code slightly different than standard
         Python. Alternatively, you can parse an expression using the
         ``'python'`` parser to retain strict Python semantics.  See the
         :ref:`enhancing performance <enhancingperf.eval>` documentation for
         more details.
-    engine : string or None, default 'numexpr', {'python', 'numexpr'}
+    engine : {'python', 'numexpr'}, optional, default 'numexpr'
```
Comment (datapythonista) [inline pandas/core/computation/eval.py]: can you remove the optional please

File: pandas/core/reshape/tile.py [hunk 13]
```diff
@@ -285,10 +285,10 @@ def qcut(x, q, labels=None, retbins=False, precision=3, duplicates="raise"):
     Parameters
     ----------
     x : 1d ndarray or Series
-    q : integer or array of quantiles
+    q : int or array of quantiles
```
Comment (datapythonista) [inline pandas/core/reshape/tile.py]: `int or list-like of int`

File: pandas/io/json/_json.py [hunk 14]
```diff
@@ -384,7 +384,7 @@ def read_json(
         By file-like object, we refer to objects with a ``read()`` method,
         such as a file handler (e.g. via builtin ``open`` function)
         or ``StringIO``.
-    orient : string,
+    orient : str,
```
Comment (datapythonista) [inline pandas/io/json/_json.py]: comma not needed

File: pandas/io/pytables.py [hunk 15]
```diff
@@ -1025,8 +1025,8 @@ def append(
                        Write as a PyTables Table structure which may perform
                        worse but allow more flexible operations like searching
                        / selecting subsets of the data
-        append       : boolean, default True, append the input data to the
-            existing
+        append       : bool, default True,
```
Comment (datapythonista) [inline pandas/io/pytables.py]: tailing comma not needed

File: pandas/io/pytables.py [hunk 16]
```diff
@@ -1025,8 +1025,8 @@ def append(
...
+            append the input data to the existing
```
Comment (datapythonista) [inline pandas/io/pytables.py]: Start with capital and end with period

File: pandas/io/pytables.py [hunk 17]
```diff
@@ -1037,8 +1037,9 @@ def append(
         chunksize    : size to chunk the writing
         expectedrows : expected TOTAL row size of this table
         encoding     : default None, provide an encoding for strings
-        dropna       : boolean, default False, do not write an ALL nan row to
-            the store settable by the option 'io.hdf.dropna_table'
+        dropna       : bool, default False,
```
Comment (datapythonista) [inline pandas/io/pytables.py]: tailing comma not needed

File: pandas/io/pytables.py [hunk 18]
```diff
@@ -1037,8 +1037,9 @@ def append(
...
+            do not write an ALL nan row to the store settable
+            by the option 'io.hdf.dropna_table'
```
Comment (datapythonista) [inline pandas/io/pytables.py]: start with capital and end with comma

**Review by datapythonista — COMMENTED** [review]
Looks great, few minor things and we're done. Thanks!

**Comment by simonjayhawkins** [comment]
Thanks @tolaa001

**Pull request merged**